File: PRSWeb/Models/Vendor.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Web;

namespace PRSWeb.Models {
	public class Vendor {
		public int Id { get; set; }

		[StringLength(12)]
		[Required]
		public string Code { get; set; }

		[StringLength(30)]
		[Required]
		public string Name { get; set; }

		[StringLength(50)]
		[Required]
		public string Address { get; set; }

		[StringLength(50)]
		[Required]
		public string City { get; set; }

		[StringLength(2)]
		[Required]
		public string State { get; set; }

		[StringLength(10)]
		[Required]
		public string Zip { get; set; }

		[StringLength(12)]
		[Required]
		public string Phone { get; set; }

		[StringLength(60)]
		[Required]
		public string Email { get; set; }

		[Required]
		public bool PreApproved { get; set; }
	}
}

File: ng/src/app/models/Vendor.ts
export class Vendor{

	Id: number;
	Code: string;
	Name: string;
	Address: number;
	City: string;
	State: string;
	Zip: number;
	Phone: string;
	Email: string;
	PreApproved: boolean;

	constructor(
		Id: number,
		Code: string,
		Name: string,
		Address: number,
		City: string,
		State: string,
		Zip: number,
		Phone: string,
		Email: string,
		PreApproved: boolean
	){
		this.Id = Id;
		this.Code = Code;
		this.Name = Name;
		this.Address = Address;
		this.City = City;
		this.State = State;
		this.Zip = Zip;
		this.Phone = Phone;
		this.Email = Email;
		this.PreApproved = PreApproved;
	}
}